idp-service/src/main/java/io/mosip/idp/services/OAuthServiceImpl.java

```diff
@@ -61,7 +55,8 @@ public class OAuthServiceImpl implements OAuthService {
 
     @Override
     public TokenResponse getTokens(@Valid TokenRequest tokenRequest) throws IdPException {
-        IdPTransaction transaction = cacheUtilService.getAuthenticatedTransaction(tokenRequest.getCode());
+        String codeHash = IdentityProviderUtil.generateB64EncodedHash(ALGO_MD5, tokenRequest.getCode());
```

Comment:
Lets change the algorithm as md5 is vulnerable.

Reply:
Done